prov/efa/src/rxr/rxr_pkt_type_data.c

```diff
@@ -304,16 +297,101 @@ int rxr_pkt_proc_data(struct rxr_ep *ep,
 
 		rxr_msg_multi_recv_free_posted_entry(ep, rx_entry);
 		rxr_release_rx_entry(ep, rx_entry);
-		return 0;
+		return;
 	}
 
 	if (!rx_entry->window) {
 		assert(rx_entry->state == RXR_RX_RECV);
-		ret = rxr_pkt_post_ctrl_or_queue(ep, RXR_RX_ENTRY, rx_entry, RXR_CTS_PKT, 0);
+		err = rxr_pkt_post_ctrl_or_queue(ep, RXR_RX_ENTRY, rx_entry, RXR_CTS_PKT, 0);
+		if (OFI_UNLIKELY(err)) {
+			FI_WARN(&rxr_prov, FI_LOG_CQ, "Cannot post CTS packet\n");
+			rxr_cq_handle_rx_error(ep, rx_entry, err);
+			rxr_release_rx_entry(ep, rx_entry);
+		}
 	}
 
 	rxr_pkt_entry_release_rx(ep, pkt_entry);
-	return ret;
+}
+
+
+/* rxr_pkt_proc_data() processes data of a DATA pkt entry or a READRSP
+ * pkt entry.
+ *
+ * It can be called by either rxr_pkt_handle_data_recv() or
+ * rxr_pkt_handle_readrsp_recv() depend on packet type.
+ *
+ * It will copy data from data pkt_entry to receiving buffer in
+ * rx_entry.
+ *
+ * There can be two cases:
+ *
+ *    1. If the receiving buffer is on host memory, this function will
+ *       call ofi_copy_to_iov() to copy data, then call
+ *       rxr_pkt_handle_eager_rtm_data_copied().
+ *
+ *    2. If the receiving buffer is on GPU memory, this function will
+ *       call rxr_pkt_post_read_to_copy_data(), which post a RDMA read
+ *       to copy data.
+ *
+ *       If the read was issued successfully,
+ *       rxr_pkt_handle_eager_rtm_data_copied()
+ *       will be called by rxr_pkt_handle_copy_by_read_completion(),
+ *       when the completion entry of the read was received.
+ *
+ *       If the read failed, an error completion entry will be written
```

Comment:
Comments like this aren't needed, it's obvious from the code.

Reply:
Removed in new revision.